src/Security/Security.php

```diff
@@ -113,419 +179,461 @@ public function createRole($id, array $content, array $options = [])
+     * @throws InvalidArgumentException
      */
-    public function createRestrictedUser($id, array $content, array $options = [])
+    public function createRestrictedUser($id, array $body, array $options = [])
     {
-        $action = 'createRestrictedUser';
-        $data = [
-            '_id' => $id,
-            'body' => $content
-        ];
+        if (empty($id) || empty($body)) {
+            throw new InvalidArgumentException('Kuzzle\Security::createRestrictedUser: Unable to create restricted user: no id or content specified');
+        }
+
+        if (empty($body['content'])) {
+            throw new InvalidArgumentException('Kuzzle\Security::createRestrictedUser: Unable to create restricted user: body["content"] is required');
+        }
+
+        if (empty($body['credentials'])) {
+             throw new InvalidArgumentException('Kuzzle\Security::createRestrictedUser: Unable to create restricted user: body["credentials"] is required');
+        }
+
+        $options['httpParams'] = [':_id' => $id];
 
         $response = $this->kuzzle->query(
-            $this->buildQueryArgs($action),
-            $data,
+            $this->kuzzle->buildQueryArgs('security', 'createRestrictedUser'),
+            [
+                '_id' => $id,
+                'body' => json_encode($body)
+            ],
             $options
         );
 
-        return new User($this, $response['result']['_id'], $response['result']['_source'], $response['result']['_meta']);
+        return new User($this->kuzzle, $response['result']['_id'], $response['result']['_source']['profileIds']);
     }
 
     /**
-     * Replaces an existing user in Kuzzle.
+     * Creates the first admin user in Kuzzle.
+     * Does nothing if an admin user already exists.
      *
      * @param integer $id Unique user identifier, will be used as username
      * @param array $content Data representing the user
      * @param array $options Optional arguments
+     *
      * @return User
+     *
+     * @throws InvalidArgumentException
      */
-    public function replaceUser($id, array $content, array $options = [])
+    public function createFirstAdmin(array $body = [], array $options = [])
     {
-        $action = 'replaceUser';
-        $data = [
-            '_id' => $id,
-            'body' => $content
-        ];
+        if (empty($body)) {
+            throw new InvalidArgumentException('Kuzzle\Security::createFirstAdmin: Unable to create first admin: no id or content specified');
+        }
+
+        if (empty($body['content'])) {
+            throw new InvalidArgumentException('Kuzzle\Security::createFirstAdmin: Unable to create first admin: body["content"] is required');
+        }
+
+        if (empty($body['credentials'])) {
+             throw new InvalidArgumentException('Kuzzle\Security::createFirstAdmin: Unable to create first admin: body["credentials"] is required');
+        }
+
+        $reset = empty($options['reset']) ? false : $options['reset'];
```

Comment:
nitpicking: we could remove $options['reset'] just after this line
(but it's not hardly required as it will just be ignored by kuzzle->query method)